testing_guide.md

```diff
@@ -0,0 +1,706 @@
+
+> :information_source: At this point you could use a low enough fee to make sure that the transaction will remain unconfirmed until you complete the next step. This is not applicable for signet.
+
+Send the transaction. Observe that the transaction is accounted as "Pending balance" and the unconfirmed output is not available as input for a new transaction.
+
+<img width="500" alt="send transaction" src="https://user-images.githubusercontent.com/18506343/190140271-c8e8e641-2933-4eaa-b27b-2cbc6d2a743a.gif">
+
+In order to test that prohibiting the spending of unconfirmed change persists to bitcoind:
+
+- Open bitcoind and use the `getbalance` RPC. The available balance that returns as a result should match what's shown in the GUI.
+- Using the GUI, re-enable the spending of unconfirmed change and observe that the available balance (that now includes the unconfirmed change) is the same between the GUI and bitcoind.
+
+> :information_source: If you are comfortable with bitcoin-cli, you could also try to create a transaction that includes the unconfirmed change.
+
+Once the test is successful, don't forget to [stop your node and clean the datadir](#4-reset-testing-environment).
+
+## Testing transient addresses for I2P outbound connections
+
+Since Bitcoin Core 22.0 a node can run over I2P. In I2P connections, the host that receives the connection knows the I2P address of the connection initiator, allowing for a number of undesirable behaviors. **Bitcoin Core 24.0 mitigates that** (if a node is not accepting incoming I2P connections) **by creating a disposable, one-time I2P address for each new outgoing connection**. For more information on I2P as well as setup instructions, see [doc/i2p.md](https://github.com/bitcoin/bitcoin/blob/master/doc/i2p.md).
+
+Add the following to your `$DATA_DIR/bitcoin.conf`(as an alternative to using cli args):
+
+```shell
+debug=i2p # to get the relevant logs in your debug.log file
+onlynet=i2p # to easily spot if there is a connectivity issue
+i2psam=127.0.0.1:7656 # the usual address and port the SAM proxy is listening to
+```
+
+We will test this change by running an I2P node that accepts incoming connections and then a node that is not accepting incoming I2P connections. **Make sure that you are able to run an I2P router with SAM proxy enabled before proceeding with this test**.
+
+> :bulb: Instead of scanning through the lengthy response of `getpeersinfo` in order to confirm your address for each outgoing connection, you can **use some `jq` magic to get a simplified view**:
+>
+> ```shell
+> cli getpeerinfo | jq 'map(select(.network == "i2p" and .inbound == false)) | map({addrbind: .addrbind})'
+> ```
+
+First, run an I2P node that is accepting incoming connections and confirm that **your I2P address is the same for all outgoing connections**.
+
+```shell
+bitcoind -i2pacceptincoming=1
+```
+
```

Comment:
As posted on the guide feedback issue  [26092](https://github.com/bitcoin/bitcoin/issues/26092) perhaps we could include this screenshot for better understanding of what to expect here:

![image](https://user-images.githubusercontent.com/110166421/195895485-cba24b04-6a91-44d8-b159-941cbbef3165.png)